### PR TITLE
Init AMQP k6 plugin

### PR DESCRIPTION
**Pull request by lxkuz**
Initial AMQP functionality including:
* Open connection
* Declare queue
* Bind/unbind queue to exchange
* Inspect queue
* Delete queue
* Declare exchange
* Delete exchange
* Bind/unbind exchange to another exchange
* Publish/listen message into the queue
* Purge queue

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,7 +2,7 @@
 >
 > As this is a proof of concept,  it won't be supported by the k6 team.
 > It may also break in the future as xk6 evolves. USE AT YOUR OWN RISK!
-> Any issues with the tool should be raised [here](https://github.com/lxkuz/xk6-amqp/issues).
+> Any issues with the tool should be raised [here](https://github.com/k6io/xk6-amqp/issues).
```
Comment (simskij) [inline README.md:5]:
```suggestion
> Any issues with the tool should be raised [here](https://github.com/grafana/xk6-amqp/issues).
```

File: README.md [hunk 2]
```diff
@@ -30,7 +30,7 @@ Then:
 
 2. Build the binary:
   ```bash
-  $ xk6 build --with github.com/lxkuz/xk6-amqp@latest
+  $ xk6 build --with github.com/k6io/xk6-amqp@latest
```
Comment (simskij) [inline README.md:33]:
```suggestion
  $ xk6 build --with github.com/grafana/xk6-amqp@latest
```

File: amqp.go [hunk 3]
```diff
@@ -1,17 +1,119 @@
...
+	exchanges := Exchanges{}
+	generalAmqp := Amqp{
+		Version:   version,
+		Queues:    &queues,
+		Exchanges: &exchanges,
+	}
+
+	modules.Register("k6/x/amqp", &generalAmqp)
+	modules.Register("k6/x/amqp/queues", &queues)
+	modules.Register("k6/x/amqp/exchanges", &exchanges)
```
Comment (simskij) [inline amqp.go:118]:
```suggestion
	modules.Register("k6/x/amqp/queue", &queues)
	modules.Register("k6/x/amqp/exchange", &exchanges)
```

Use singularis throughout unless the pluralization adds semantic value.

File: exchanges.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+type Exchanges struct {
+	Version    string
+	Connection *amqpDriver.Connection
+}
+
+type ExchangeOptions struct {
+	ConnectionUrl string
+}
+
+type EchangeDeclareOptions struct {
```
Comment (simskij) [inline exchanges.go:16]:
```suggestion
type ExchangeDeclareOptions struct {
```

**Review by simskij — CHANGES_REQUESTED** [review]
Minor feedback items. All in all; looks good. 👏🏽

File: README.md (hunk 1 above)
Comment (lxkuz) [reply README.md:5]: @simskij does it mean I should use `github.com/grafana` instead of `github.com/k6io` **everywhere**?

**Comment by lxkuz** [comment]
@simskij review comments handled

File: README.md (hunk 1 above)
Comment (simskij) [reply README.md]: Yes, correct

File: README.md [hunk 5]
```diff
@@ -45,9 +45,48 @@ $ ./build.sh && ./k6 run my-test-script.js
...
+  const queueName = 'K6 general'
+  
+  Queues.declare({
+    name: queueName,
+    // durable: false,
+    // delete_when_unused: false,
+    // exclusive: false,
+    // no_wait: false,
+    // args: null
+  })
```
Comment (simskij) [inline README.md:67]: Singular form of `Queue`

File: queues.go [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+}
+
+func (queues *Queues) Purge(name string, noWait bool) (int, error) {
+	ch, err := queues.Connection.Channel()
+	if err != nil {
+		return 0, err
+	}
+	defer ch.Close()
+	return ch.QueuePurge(name, noWait)
+}
```
Comment (simskij) [inline queues.go:116]: Singular form of `Queue`

File: exchanges.go [hunk 7]
```diff
@@ -0,0 +1,100 @@
+package amqp
```
Comment (simskij) [inline exchanges.go:1]: Singular form of `Exchange`

**Review by simskij — CHANGES_REQUESTED** [review]

**Pull request merged**